### DISCUSSION TITLE
Any way to use DispatchProxy v6.0.0 from a net48 target?

### DISCUSSION THREAD

**Discussion by menees** [dotnet/runtime#64726] [Q&A] [answered]
The [System.Reflection.DispatchProxy NuGet package](https://www.nuget.org/packages/System.Reflection.DispatchProxy/) was never updated for the final 5.0.0 or 6.0.0 version. But I'd really like to reference the 6.0.0 package in a project that multi-targets .NET Framework 4.8 and .NET 6.0. The net6.0 build just works (presumably due to the Microsoft.NET.Sdk), but to get the net48 build to compile I have to reference the older, buggy v4.7.1 package like this:

``` XML
<Choose>
  <When Condition=" '$(TargetFramework)' == 'net48' ">
    <ItemGroup>
      <PackageReference Include="System.Reflection.DispatchProxy" Version="4.7.1" />
    </ItemGroup>
  </When>
</Choose>
```

Is there a way to reference System.Reflection.DispatchProxy v6.0.0 for net48? I ask because it contains the fix for bug #30917. In my net48 build I have to make my client proxy class public with a hack like this in order for the v4.7.1 version to work:

``` C#
#if NETFRAMEWORK
public
#else
internal
#endif
class ClientProxy : DispatchProxy
```

I don't like leaking that implementation detail from my project. If there's no way to reference the SDK's v6.0.0 library from net48, is there any chance the DispatchProxy NuGet package version will ever be updated to a final 6.0.0 version?

**Comment by huoyaoyuan** [comment, marked as answer]
It seems that the OOB version of that package has been discontinued. See #39220.
I'm not sure how to deliver bug fix for it in this scenario. Asking @ericstj

**Reply by ericstj** [reply]
It looks like the fix for #30917 also required and runtime changes https://github.com/dotnet/runtime/pull/47784 so it might not be feasible if those changes were necessary in .NETFramework.  cc @davidwrighton 

If this only required changes to the package then, the last release which shipped the package was [3.1](https://github.com/dotnet/corefx/tree/release/3.1/src/System.Reflection.DispatchProxy).  So a request could be made to backport a fix to that release: and it would need to be evaluated against the bar for that release.  Changes to .NETFramework need to go through https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-framework and the bar is much higher.

As a workaround I'd recommend to hide that type from intellisense with `[EditorBrowsable(EditorBrowsableState.Never)]`

**Reply by menees** [reply]
Thanks! Since it requires runtime changes, it's not worth trying to get it backported to .NET Framework. The EditorBrowsable workaround will be fine since it's minimal pain for everyone.
